### PR TITLE
oc-mail: Added [MAPIStoreFolder getPidTagDisplayName:inMemCtx:]

### PR DESCRIPTION
**Pull request by javierag**
This getter is necessary to decode folder names
in utf7 encoding.

Suggested message for NEWS:
- Non-latin folder names are displayed correctly on Outlook

### CODE REVIEW

**Comment by sixstone-qq** [comment]
This change must be in MAPIStoreMailFolder as it is only required for IMAP folders, right?

**Comment by javierag** [comment]
Yes, you are right I have fumbled this one. I wil fix this

**Comment by javierag** [comment]
Fxied. Sorry for the noise.

File: OpenChange/MAPIStoreMailFolder.m [hunk 1]
```diff
@@ -257,6 +257,16 @@ - (int) getPidTagContainerClass: (void **) data
   return MAPISTORE_SUCCESS;
 }
 
+
+- (int) getPidTagDisplayName: (void **) data
+                    inMemCtx: (TALLOC_CTX *) memCtx
+{
+  NSString *displayName = [[sogoObject displayName] stringByDecodingImap4FolderName];
```
Comment (blaxter) [inline OpenChange/MAPIStoreMailFolder.m]: I think we should first `fromCSSIdentifier` and then `stringByDecodingImap4FolderName`

**Comment by blaxter** [comment]
I'm not sure if this is the best way, I'd try to fix this in SOGoMailFolder.m displayName so it returns the name already properly decoded

**Comment by javierag** [comment]
I will try your way @blaxter . Anyway a double utf7 decoding should not break anything so even if someone is expecting a utf for [SOGoMailFolder displayName] it should be harmless.

**Comment by javierag** [comment]
Done. I havent blend the two commits to be able to revert quickly if we want to return to the previous implementation.
If it all ok, I will do it before the merge.

**Comment by papajulio** [comment]
:+1: Tested and working fine.

**Comment by blaxter** [comment]
This works for subfolders but there are still issues when creating root folders. But let's merge this :+1:

**Pull request merged**